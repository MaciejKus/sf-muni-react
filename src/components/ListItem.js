import React from 'react';
import PropTypes from 'prop-types';
import '../styles/ListItems.css';

class ListItem extends React.Component {
  constructor() {
    super();
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    if (this.props.isActive) {
      this.props.handleRemoveRoute(this.props.tag);
    } else {
      this.props.handleAddRoute(this.props.tag);
    }
  }

  render() {
    return (
      <li
        onClick={this.handleClick}
        className={this.props.isActive ? 'isActive' : ''}
      >
        {this.props.title}
      </li>
    );
  }
}

ListItem.defaultProps = {
  isActive: false,
};

ListItem.propTypes = {
  isActive: PropTypes.bool,
  handleAddRoute: PropTypes.func.isRequired,
  handleRemoveRoute: PropTypes.func.isRequired,
  tag: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default ListItem;
