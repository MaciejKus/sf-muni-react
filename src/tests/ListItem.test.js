import React from 'react';
import ListItem from '../components/ListItem';
import renderer from 'react-test-renderer';

describe('ListItem', () => {
  let props;

  beforeEach(() => {
    props = {
      isActive: false,
      handleAddRoute: () => {},
      handleRemoveRoute: () => {},
      tag: '3',
      title: '3 bus line',
    };
  })

  test('Matches snapshot', () => {
    const component = renderer.create(
      <ListItem
       {...props}
      />    
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    props.isActive = true;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
