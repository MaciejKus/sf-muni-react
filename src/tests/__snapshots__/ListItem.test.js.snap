// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Matches snapshot 1`] = `
<li
  className=""
  onClick={[Function]}
>
  3 bus line
</li>
`;

exports[`ListItem Matches snapshot 2`] = `
<li
  className=""
  onClick={[Function]}
>
  3 bus line
</li>
`;
